File: attempt1/generateSheet.m
%% Step 0:
% Set up environment and variables
clear
load("DraftsAndStats.mat")
% Uncomment the following line to switch to the testing data set
TrainDraft = train;
%TrainDraft = DraftPrelim;

%% Step 1:
% Take Draft Year, Made NBA, Draft Position, NBA Performance Rating, Name
% from trainList
draftList = TrainDraft(:, 23);
madeNBA = ones(height(TrainDraft), 1);
draftPosition = TrainDraft(:, 2);
name = TrainDraft(:, 4);
NBAPR = TrainDraft(:, 21);

%% Step 2:
% Match name with CollegeStats name, get College WS, College Adjusted BPM, and
% College PER

nameStatsList = table2array(CollegeStats(:, 1));
firstLetterIndexList = [1];
lastLetter = "A";
for j = 1:height(CollegeStats)
    currentName = char(CollegeStats{j, 1});
    firstLetter = currentName(1);
    if (firstLetter > lastLetter)
        lastLetter = firstLetter;
        firstLetterIndexList = [firstLetterIndexList ; j];
    end
end
firstLetterIndexList = [firstLetterIndexList ; height(CollegeStats)]

collegeWS = [];
collegeBPM = [];
collegePER = [];

currentLetter = "A";
letterIndex = 1;
letterArray = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'];
for i = 1:height(name)
    display(i)
    seasonsList = [];
    
    currentName = char(name{i, 1});
    currentLetter = currentName(1);
    while (currentLetter > letterArray(1, letterIndex))
        letterIndex = letterIndex + 1;
    end
    
    for j = firstLetterIndexList(letterIndex, 1):(firstLetterIndexList(letterIndex + 1, 1))
        if (strcmp(char(CollegeStats{j, 1}), currentName))
            seasonsList = [seasonsList ; CollegeStats(j, :)];
        end
    end
    
    display(seasonsList)
    if (~isempty(seasonsList))
        bestLine = seasonsList(1, :);
        for j = 2:height(seasonsList)
            if (seasonsList{j, 6} > bestLine{1, 6})
                bestLine = seasonsList(j, :);
            end
        end
   
        collegeWS = [collegeWS ; bestLine{1, 20}];
        collegeBPM = [collegeBPM ; bestLine{1, 40}];
        collegePER = [collegePER ; bestLine{1, 36}];
    else
        collegeWS = [collegeWS ; -1];
        collegeBPM = [collegeBPM ; -1];
        collegePER = [collegePER ; -1];
    end
end

collegeWS = array2table(collegeWS);
collegeBPM = array2table(collegeBPM);
collegePER = array2table(collegePER);
madeNBA = array2table(madeNBA);

tempFinalTable = [draftList madeNBA draftPosition NBAPR name collegeWS collegeBPM collegePER];

%% Step 3 (add after preliminary results):
% Match name with FinalPlayerInfo name, get Position, Scouting Report,
% Height, Wingspan, Age

File: attempt1/generateSheet.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr  5 03:11:47 2019

@author: angelo
"""

import pandas as pd

train_data = "C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\DraftPrelims\\2003-2013DraftPrelim.xlsx"
#test_data = "C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\DraftPrelims\\2014-2015DraftPrelim.xlsx"
#player_info = "C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\FinalPlayerInfo.xlsx"
#player_stats = "C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\FinalBPM.xlsx"
output_sheet = "C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\outputSheet.xlsx"

NUM_ROWS_TRAIN = 448
NUM_ROWS_TEST = 83
NUM_ROWS_INFO = 7669
NUM_ROWS_STATS = 22808

trainList = pd.read_excel(train_data)
#testList = pd.read_excel(test_data)
#infoList = pd.read_excel(player_info)
#statsList = pd.read_excel(player_stats)

print(trainList)
print(trainList.iloc[1, 1])
#print(trainList.iloc[0, 1])
#print(trainList.iloc[0, 20])
#print(trainList.iloc[0, 3])

# Step 1:
# Take Draft Year, Made NBA, Draft Position, NBA Performance Rating, Name
# from trainList
for i in range(NUM_ROWS_TRAIN):
    i += 1
    output_sheet.iloc[i, 0] = trainList.iloc[i, 22]
    output_sheet.iloc[i, 1] = 1
    output_sheet.iloc[i, 2] = trainList.iloc[i, 1]
    output_sheet.iloc[i, 3] = trainList.iloc[i, 20]
    output_sheet.iloc[i, 4] = trainList.iloc[i, 3]

# Step 2:
# Match name with FinalBPM name, get College WS, College Adjusted BPM, and
# College PER


# Step 3 (add after preliminary results):
# Match name with FinalPlayerInfo name, get Position, Scouting Report,
# Height, Wingspan, Age

"""
removeCount = 0
for i in range(NUM_ROWS_ENTRY):
    i -= removeCount
    if not pd.notna(entryList.iloc[i, 15]):
        entryList.drop(entryList.index[i], inplace=True)
        removeCount += 1

entryList.to_excel("C:\\Users\\angel\\OneDrive\\College\\Work\\ESSA\DataEntryWithCEEBs.xlsx")

entryList = pd.read_excel("C:\\Users\\angel\\OneDrive\\College\\Work\\ESSA\\IBDPandCounselorsDataEntry.xlsx")

removeCount = 0
for i in range(NUM_ROWS_ENTRY):
    i -= removeCount
    if pd.notna(entryList.iloc[i, 15]):
        entryList.drop(entryList.index[i], inplace=True)
        removeCount += 1

"""

statsList.to_excel("C:\\Users\\angel\\OneDrive - The College of Wooster\\College\\Semester2\\Math Modelling\\Final Project\\outputSheet.xlsx")